### PR TITLE
Feature/admin account

### PR DESCRIPTION
**Pull request by Adrie0291**
- created an admin account using roles
- passy login: admin2@o2.pl  password: haslo
- adding an admin account using user roles

### CODE REVIEW

File: src/main/java/com/sda/eventapp/web/mvc/controller/EventController.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public String update(ModelMap model, @PathVariable Long id) {
         model.addAttribute("eventImage", eventToUpdate.getImage());
 
         User loggedUser = (User) authenticationFacade.getAuthentication().getPrincipal();
-        if (loggedUser.getId() != (eventService.findOwnerIdByEventId(id))) {
+        if (loggedUser.getId() != (eventService.findOwnerIdByEventId(id)) && !loggedUser.getRole().equals(User.Roles.ROLE_ADMIN)) {
```
Comment (JasinskiSz) [inline src/main/java/com/sda/eventapp/web/mvc/controller/EventController.java:83]: There is a logic error. Should be `||` instead of `&&`.
Right now we'll get inside of this if only when logged user is an owner. The other check will be skipped.

File: src/main/java/com/sda/eventapp/service/UserService.java [hunk 2]
```diff
@@ -18,13 +18,13 @@ public class UserService {
...
         User user = mapper.toUser(form);
         user.setPassword(passwordEncoder.encode(form.getPassword()));
+        user.setRole(User.Roles.ROLE_USER);
         return repository.save(user);
     }
 
     public boolean existsByEmail(String email) {
         return repository.existsByEmail(email);
     }
-
```
Comment (JasinskiSz) [inline src/main/java/com/sda/eventapp/service/UserService.java:27]: This here also shouldn't be removed.

File: src/main/resources/templates/event-detail-view.html [hunk 3]
```diff
@@ -153,8 +153,8 @@ <h1 class=" text-break display-1 fw-bolder text-white mb-2" th:text="${event.tit
                         <div sec:authorize="isAuthenticated()">
                             <div th:if="${event.startingDateTime.isAfter(#temporals.createNow())}">
                                 <!--Update event if Owner and If Future-->
-                                <div th:if="${#strings.equals(event.ownerNickname, loggedUser.username)}">
-                                    <a class="btn btn-success btn-lg px-4 me-sm-3"
+                                <div th:if="${#strings.equals(event.ownerNickname, loggedUser.username) || loggedUser.role.toString().equals('ROLE_ADMIN')}">
+                                    <a class=" btn btn-success btn-lg px-4 me-sm-3"
```
Comment (JasinskiSz) [inline src/main/resources/templates/event-detail-view.html:157]: Please, don't add unnecessary changes to our code. This space changes nothing so please remove it.

File: src/main/java/com/sda/eventapp/EventAggregationAppApplication.java [hunk 4]
```diff
@@ -1,13 +1,32 @@
...
 
+import com.sda.eventapp.model.User;
+import com.sda.eventapp.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.security.crypto.password.PasswordEncoder;
 
+@RequiredArgsConstructor
```
Comment (JasinskiSz) [inline src/main/java/com/sda/eventapp/EventAggregationAppApplication.java:11]: I'm getting a feeling that this shouldn't be in this class. This is just to start Spring boot, but if I am mistaken, please inform me.

**Review by JasinskiSz — CHANGES_REQUESTED** [review]

File: src/main/java/com/sda/eventapp/web/mvc/controller/EventController.java (hunk 1 above)
Comment (Adrie0291) [reply src/main/java/com/sda/eventapp/web/mvc/controller/EventController.java:83]: There is no logic error. The && operator is correct. 
The first condition checks if the id of the logged in user **is not** equal to the id of the event owner.
 And the second condition checks if the logged in user **has no** admin role.
 If both are true it will throw 403
 
 I can changed "!loggedUser.getRole().equals(User.Roles.ROLE_ADMIN)" to "loggedUser.getRole() != User.Roles.ROLE_ADMIN" (same but looks better). What do you think?

File: src/main/resources/templates/event-detail-view.html (hunk 3 above)
Comment (Adrie0291) [reply src/main/resources/templates/event-detail-view.html:157]: this code is a need as it is responsible for displaying button the Update Event on the page...

File: src/main/java/com/sda/eventapp/EventAggregationAppApplication.java (hunk 4 above)
Comment (Adrie0291) [reply src/main/java/com/sda/eventapp/EventAggregationAppApplication.java:11]: Honestly, I do not know. I need this Annotation to create constructor user repository and passwordencoder. But i can changed place and i can create admin in UserService. For example User Service implements CommandLineRunner ...
 What do you think?

File: src/main/resources/templates/event-detail-view.html (hunk 3 above)
Comment (JasinskiSz) [reply src/main/resources/templates/event-detail-view.html:157]: I'm referring to the space in line 157.

File: src/main/java/com/sda/eventapp/web/mvc/controller/EventController.java (hunk 1 above)
Comment (JasinskiSz) [reply src/main/java/com/sda/eventapp/web/mvc/controller/EventController.java:83]: Yup, I think you're right!
I don't think there is need to change anything else here

Comment (JasinskiSz) [reply src/main/java/com/sda/eventapp/web/mvc/controller/EventController.java:83]: Actually... You can just remove these not necessary parentheses.

**Comment by JasinskiSz** [comment]
Continued on the PR #47
